### ISSUE TITLE
Fail to run tflite on DSP - BATCH_TO_SPACE_ND failed to prepare

### ISSUE THREAD

**Issue by shlomi-amitai** [tensorflow/tensorflow#45478]
@tensorflow/micro

**System information**
- Host OS Platform and Distribution (e.g., Linux Ubuntu 16.04):
Android 

- TensorFlow installed from (source or binary):
source (dde0fe783c8208358971979460611caca75dc363)

- Tensorflow version (commit SHA if source):
tf-nightly 2.4

- Target platform (e.g. Arm Mbed OS, Arduino Nano 33 etc.):
dsp on mobile

**Describe the problem**
INFO: Created TensorFlow Lite delegate for Hexagon.
INFO: Initialized TensorFlow Lite runtime.
INFO: Interpreter::UseNNAPI() is deprecated. Use tflite::NnApiDelegate() directly instead.
INFO: Hexagon delegate: 52 nodes delegated out of 88 nodes with 10 partitions.

ERROR: tensorflow/lite/kernels/batch_to_space_nd.cc:81 output_batch_size % block_shape[dim] != 0 (1 != 0)
ERROR: Node number 6 (BATCH_TO_SPACE_ND) failed to prepare.

**Please provide the exact sequence of commands/steps when you ran into the problem**
tflite model attached
[myModel4DSP.zip](https://github.com/tensorflow/tensorflow/files/5660174/myModel4DSP.zip)

**Comment by thaink** [comment]
Seems like a Micro issue.

**Comment by advaitjain** [comment]
From the error message, it looks like the Lite (not Micro) kernels are being use.

@shlomi-amitai can you please confirm if you are looking to run on the Android CPU + delegates (in which case this would be a Tensorflow Lite Mobile issue) or on an embedded DSP (in which case this would be a TensorflowLite Micro issue).

**Comment by shlomi-amitai** [comment]
Hi, I'm trying to run a quantized tflite model on an android mobile DSP.
below listed conversion's params (I use IOQ):
    converter.optimizations = [tf.lite.Optimize.DEFAULT] # dynamic range quantization
    if quantType == 'ffq' or quantType == 'ioq':
        converter.representative_dataset = representative_data_gen  # + float fallback quantization
        if quantType == 'ioq':
            converter.target_spec.supported_ops = [tf.lite.OpsSet.TFLITE_BUILTINS_INT8]  # +integer only quantization
            converter.inference_input_type = tf.int8 # +integer only quantization
            converter.inference_output_type  = tf.int8

**Comment by shlomi-amitai** [comment]
Solved.
For those who might meet this issue.
Input batch size should be changed to the exact number (e.g. 1)  instead of None (as when training).
